gpx.go

```diff
@@ -466,34 +473,44 @@ func (w *WptType) UnmarshalXML(d *xml.Decoder, start xml.StartElement) error {
 // MarshalXML implements xml.Marshaler.MarshalXML.
 func (g *GPX) MarshalXML(e *xml.Encoder, start xml.StartElement) error {
 	baseURL := "http://www.topografix.com/GPX/" + strings.Join(strings.Split(g.Version, "."), "/")
+	attr := []xml.Attr{
+		{
+			Name:  xml.Name{Local: "version"},
+			Value: g.Version,
+		},
+		{
+			Name:  xml.Name{Local: "creator"},
+			Value: g.Creator,
+		},
+		{
+			Name:  xml.Name{Local: "xmlns:xsi"},
+			Value: "http://www.w3.org/2001/XMLSchema-instance",
+		},
+		{
+			Name:  xml.Name{Local: "xmlns"},
+			Value: baseURL,
+		},
+		{
+			Name:  xml.Name{Local: "xsi:schemaLocation"},
+			Value: baseURL + " " + baseURL + "/gpx.xsd " + g.XMLSchemaLoc,
```

Comment:
The trailing space at the end of `"gpx.xsd "` is included in `Value` even if `g.XMLSchemaLoc` is empty. This breaks several tests. Please change this so that there is no trailing space in `Value` when g.XMLSchemaLoc is empty.